setup.py

```diff
@@ -19,7 +19,7 @@
     "check-manifest>=0.25",
     # coverage pinned because of https://github.com/nedbat/coveragepy/issues/716
     "coverage>=4.0,<5.0.0",
-    "isort>=4.3",
+    "isort>=5.0,<6.0",
```

Comment:
I agree, thought I have mixed feelings about the upper bound:

- pro: we avoid breaking changes
- con: we need to proactively check/be alerted of new releases and do the work. otherwise we simply get them
- con: we would be blocking other people using the module in case that for some reason they need a higher version

Fixing would happen in any case. As long as we upper pin in the instance in order no to affect users...

I think the architects are gonna talk about this in the next meeting

Reply:
> pro: we avoid breaking changes

I definitely think my reaction is out of dependency fatigue (really frustration :tired_face:  ): 
- we spend an inordinate amount of time reacting to upstream dependencies
- our code base is very brittle because of them

Just in recent memory we had/have to deal  with: pipenv, werkzeug (and all libraries it depended on), marshmallow, webargs, now isort, and others not to mention my first contribution to the invenio project was doing a [fix for a new version of pytest](https://github.com/inveniosoftware/pytest-invenio/pull/19) >_< .

I don't see any other way than to capping those dependencies to the next major version.   

> con: we need to proactively check/be alerted of new releases and do the work. otherwise we simply get them

I humbly think invenio sprints would be the time to do this formally: assess library updates and their impact and incorporate them. And this is basically what invenio sprints are anyway but for internal modules, so it's fitting.

> con: we would be blocking other people using the module in case that for some reason they need a higher version

Yes, may happen :(. In this case the dependency is in `tests_require`, so it wouldn't be a problem (don't get `tests_require` from other modules). Overall, suddenly requiring a new major breaking version would be rare since invenio sprints would have done the assessments.

Reply:
Hmm I still have mixed feelings. However, since this affects global Invenio way of working. I.e. planning sprints and what for (AFAIK we do some dependency battling, but it's not what they are meant for), and how to handle dependencies on a bigger picture. I think is more up to the architects than us.

For this module, feel free to leave it pinned. However, before start pinning other modules in the Invenio code base I would wait to see what they say.